### ISSUE TITLE
Question about code

### ISSUE THREAD

**Issue by massawuyh** [lmssdd/TPNILM#2]
thr = 0.5
for i in range(20):
    #filename = '/content/gdrive/My Drive/NILM/UKDALE_seen_%d.pth' %i
    filename = './UKDALE_seen_%d.pth' %i
    print(filename)
    model.load_state_dict(torch.load(filename))
    for a in range(3):
        #x_true, p_true, s_true, s_hat = evaluate_activation(model, dl_house_total[0], a)
        #pm = p_true.sum() / s_true.sum()
        #pm = (ds_appliance[0][APPLIANCE[a]] * 
        #      ds_status[0][APPLIANCE[a]]).sum() / ds_status[0][APPLIANCE[a]].sum() / MAX_POWER
        pm = ds_appliance[0][APPLIANCE[a]].sum() / ds_status[0][APPLIANCE[a]].sum() / MAX_POWER
        x_true, p_true, s_true, s_hat = evaluate_activation(model, dl_house_test[0], a)
        s_hat = get_status(s_hat, thr, MIN_OFF[a], MIN_ON[a])
        p_hat = pm * s_hat
        scores[a]['F1'].append(f1_score(s_true, s_hat))
        scores[a]['Precision'].append(precision_score(s_true, s_hat))
        scores[a]['Recall'].append(recall_score(s_true, s_hat))
        scores[a]['Accuracy'].append(accuracy_score(s_true, s_hat))
        scores[a]['MCC'].append(matthews_corrcoef(s_true, s_hat))
        scores[a]['MAE'].append(mean_absolute_error(p_true, p_hat)*MAX_POWER)
        scores[a]['SAE'].append((p_hat.sum() - p_true.sum()) / p_true.sum())

I'm not quite understand that 
 pm = ds_appliance[0][APPLIANCE[a]].sum() / ds_status[0][APPLIANCE[a]].sum() / MAX_POWER
and 
p_hat = pm * s_hat

p_hat is from pm, but pm is from the appliance actual value? but we suppose don't know appliance actual value.
appliance value should be evaluated from model right?

**Comment by lmssdd** [comment]
Yes, but it’s part pf the standard data for any appliance. In Europe the manufacturer has to declare mean consumption per cycle and its duration, hence the average power.Best,LucaInviato da iPhoneIl giorno 12 mag 2023, alle ore 16:11, massawuyh ***@***.***> ha scritto:﻿
thr = 0.5
for i in range(20):
#filename = '/content/gdrive/My Drive/NILM/UKDALE_seen_%d.pth' %i
filename = './UKDALE_seen_%d.pth' %i
print(filename)
model.load_state_dict(torch.load(filename))
for a in range(3):
#x_true, p_true, s_true, s_hat = evaluate_activation(model, dl_house_total[0], a)
#pm = p_true.sum() / s_true.sum()
#pm = (ds_appliance[0][APPLIANCE[a]] *
#      ds_status[0][APPLIANCE[a]]).sum() / ds_status[0][APPLIANCE[a]].sum() / MAX_POWER
pm = ds_appliance[0][APPLIANCE[a]].sum() / ds_status[0][APPLIANCE[a]].sum() / MAX_POWER
x_true, p_true, s_true, s_hat = evaluate_activation(model, dl_house_test[0], a)
s_hat = get_status(s_hat, thr, MIN_OFF[a], MIN_ON[a])
p_hat = pm * s_hat
scores[a]['F1'].append(f1_score(s_true, s_hat))
scores[a]['Precision'].append(precision_score(s_true, s_hat))
scores[a]['Recall'].append(recall_score(s_true, s_hat))
scores[a]['Accuracy'].append(accuracy_score(s_true, s_hat))
scores[a]['MCC'].append(matthews_corrcoef(s_true, s_hat))
scores[a]['MAE'].append(mean_absolute_error(p_true, p_hat)*MAX_POWER)
scores[a]['SAE'].append((p_hat.sum() - p_true.sum()) / p_true.sum())
I'm not quite understand that
pm = ds_appliance[0][APPLIANCE[a]].sum() / ds_status[0][APPLIANCE[a]].sum() / MAX_POWER
and
p_hat = pm * s_hat
p_hat is from pm, but pm is from the appliance actual value? but we suppose don't know appliance actual value.
appliance value should be evaluated from model right?

**Comment by massawuyh** [comment]
pm = ds_appliance[0][APPLIANCE[a]].sum() / ds_status[0][APPLIANCE[a]].sum() / MAX_POWER

this is calculating the whold dataset (include training and testing) average consumption right?
is it possible only use the training average consumption? any big difference?

**Comment by lmssdd** [comment]
Sure, you can do it and it shouldn't make any difference.
if you have the value from the performance sheet declared by the
manufacturer it is even better,

**Comment by massawuyh** [comment]
thank you !
